shm/src/cargo.cpp

```diff
@@ -0,0 +1,10 @@
+#include "cargo.hpp"
+
+Cargo::Cargo(std::string name, size_t amount, double basePrice)
+    : name_(name),
+      amount_(amount),
+      basePrice_(basePrice) {}
+
+bool Cargo::operator==(Cargo& cargo) const {
+    return name_ == cargo.name_;
+}
```

Comment:
Redundant. This could be pure virtual, because You never create instance of Cargo